cogs/channel_mods.py

```diff
@@ -875,116 +875,95 @@ async def modlog_delete(self, ctx, user, *, index):
-                        n += 1  # (For every deleted log, the next ones will be shifted by -1. Therefore,
-                                # add 1 every time a log gets deleted to counter that.)
-                        removed_indexes.append(index)  # for every successfully deleted log, append
-                                                       # the index to the corresponding 'removed_indexes' list.
-                    elif int(index) <= 0:  # If the index is equal to or smaller than zero (forbidden values)...
-                        invalid_indexes.append(index)  # append it to the 'invalid_indexes' list.
-                except IndexError:  #  If the value is legal but the corresponding log wasn't found in config...
-                    not_found_indexes.append(index)  # append the index to the 'not_found_indexes' list.
+        else:  # -all was not called, and so there are specific indices to delete:
+            def invalid_indices_check(check_index):
+                """For indices that are strings or not greater than 0"""
+                if not check_index.isdigit():  # is not a digit
+                    return True
+                else:  # now is a digit
+                    if int(check_index) < 1:  # if index is 0 or negative
+                        return True
+
+            def not_found_check(check_index, config):
+                """For indices not in config"""
+                if check_index.isdigit():
+                    if int(check_index) >= len(config):
+                        return True  # called index is out of range of list
+
+            def indices_check(check_index, not_found):
+                if check_index.isdigit() and check_index not in not_found:
+                    if int(check_index) >= 1:
+                        return True
+
+            invalid_indices: list = [i for i in indices
+                                     if invalid_indices_check(i)]  # list of non-digit arguments
+            not_found_indices: list = [i for i in indices
+                                       if not_found_check(i, config)]  # list of indices not found in modlog
+            indices: list = [i for i in indices
+                             if indices_check(i, not_found_indices)]  # list of valid arguments
+            removed_indices: list = []  # eventual list of modlog entries successfully removed
+            indices.sort(key=int)  # Sort it numerically
 
+            n = 1
+            for index in indices:  # For every index in the list...
+                del config[int(index) - n]  # delete them from the user's modlog.
+                n += 1  # For every deleted log, the next ones will be shifted by -1. Therefore,
+                # add 1 every time a log gets deleted to counter that.
+                removed_indices.append(index)  # for every successfully deleted log, append
+                # the index to the corresponding 'removed_indexes' list.
```

Comment:
It's all just in one giant blob haha, it's probably better to view this without the highlighting